### PR TITLE
Add apple-m1/apple-m2 support

### PR DESCRIPTION
**Pull request by krenzland**
- Use longdouble instead of float128
- Add apple-m1, apple-m2
- Rename no prefetch option for LIBXSMM_JIT

### CODE REVIEW

File: yateto/gemm_configuration.py [hunk 1]
```diff
@@ -303,6 +303,8 @@ def __init__(self, arch):
       'knl' : [libxsmm_jit, libxsmm, pspamm, mkl, blis, eigen],
       'skx' : [libxsmm_jit, libxsmm, pspamm, mkl, blis, eigen],
       'thunderx2t99' : [libxsmm_jit, pspamm, openblas, blis, eigen],
+      'apple_m1' : [libxsmm_jit, pspamm, openblas, blis, eigen],
```
Comment (davschneller) [inline yateto/gemm_configuration.py:306]: Underscores here, but not anywhere else

**Review by davschneller — COMMENTED** [review]
Almost good, but I'm still sceptic

**Comment by davschneller** [comment]
However, even if float128 does only give extra padding—why don't we leave it like that? Is there a disadvantage on e.g. Apple M1/M2?

**Comment by krenzland** [comment]
Float128 is not supported on all architectures. And it gets weirder. See:
https://stackoverflow.com/a/17023995

**Comment by davschneller** [comment]
> Float128 is not supported on all architectures. And it gets weirder. See: https://stackoverflow.com/a/17023995

I know, that's why I had asked about the "extra padding". But ok, we can go be `longdouble`.

**Review by davschneller — APPROVED** [review]

**Pull request merged**